web_security_dialog/static/src/js/web_security_dialog.js

```diff
@@ -0,0 +1,107 @@
+    var Model = require('web.DataModel');
+    var framework = require('web.framework');
+
+    var QWeb = core.qweb;
+    var _t = core._t;
+
+    FromWidget.WidgetButton.include({
+        init : function(field_manager, node) {
+            this._super(field_manager, node);
+            if(this.node.attrs.options) {
+                this.options = JSON.parse(this.node.attrs.options)
+                    ? JSON.parse(this.node.attrs.options)
+                            : false;
+                if(this.options) {
+                    this.is_dialog_security = this.options.security
+                    ? this.options.security
+                            : false;
+                }else{
+                    this.is_dialog_security = false;
+                }
+            }
+        },
+        on_click : function() {
+            var self = this;
+            if (this.view.is_disabled) {
+                return;
+            }
+            self.view.enable_button();
+            self.force_disabled = false;
+            self.check_disable();
+            if (self.$el.hasClass('o_wow')) {
+                self.show_wow();
+            }
+            if (this.is_dialog_security) {
+                self.dialog = new Dialog(self,{
+                    title: _t('Security'),
+                    size : "small",
+                    $content: QWeb.render('DialogSecurity'),
+                    buttons: [
+                                {
+                                    text:_t('Cancel'),
+                                    close:true
+                                },
+                                {
+                                    text: _t("Ok"),
+                                    click: function(){
+                                       var curr_obj = this;
+                                       var pass_value = this.$el.find("#d_security").val();
+                                       if (pass_value) {
+                                           framework.blockUI();
+                                           var callback = self.validate_user(self.is_dialog_security,pass_value);
+                                           callback.done(function(result) {
+                                               framework.unblockUI();
+                                               if (result) {
+                                                   curr_obj.close();
+                                                   self.click_operation();
+                                               }else {
+                                                   Dialog.alert(self, _t("Password is Wrong"));
+                                                   return;
+                                               }
+                                           }).fail(function(error) {
+                                               framework.unblockUI();
+                                               Dialog.alert(self, _t("Connection lost"));
```

Comment:
The message should be 'Either the password is wrong or the connection is lost! Contact your Administrator.'